distributed/protocol/numba.py

```diff
@@ -18,9 +18,20 @@ def serialize_numba_ndarray(x):
 @cuda_deserialize.register(numba.cuda.devicearray.DeviceNDArray)
 def deserialize_numba_ndarray(header, frames):
     (frame,) = frames
+    strides = header["strides"]
+
+    # Starting with __cuda_array_interface__ version 2, strides can be None,
+    # meaning the array is C-contiguous, so we have to calculate it.
+    if strides is None:
+        itemsize = np.dtype(header["typestr"]).itemsize
+        strides = list(header["shape"][1:]) + [itemsize]
+        for i in range(len(strides) - 1, 0, -1):
+            strides[i - 1] *= strides[i]
```

Comment:
You might want something like the following?

```python
strides = np.cumprod(shape[::-1]) * itemsize
```

Reply:
Why didn't this show up in tests?  Don't we test with a C-contiguous array?

Reply:
Thanks for the suggestion, it's indeed what I wanted and couldn't think of when I wrote the code.

There are two reasons it didn't show up in the tests:

1. CUDA code is not being tested by CI;
1. Numba >= 0.46.0 is required (which contains version 2 of `__cuda_array_interface__`.